### PR TITLE
feat: added to guidance page and improved code snippet examples to sh…

### PR DESCRIPTION
**Pull request by za2201**
## Summary of the changes

Added usage of tooltip to clarify what a tooltip is on the guidance page. 
Added component examples to code based on user  feedback confusion regarding placement of elements. Included the buttons in code examples already given and created an example with a chip.

## Related issue

#333 

## Checklist

- [y] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [y] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Comment by MI6-255** [comment]
Will need to rebase this PR off develop, let us know if you need a hand

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by jd239 — APPROVED** [review]

File: src/content/structured/components/tooltips/code.mdx [hunk 1]
```diff
@@ -126,3 +142,52 @@ export const snippetsRight = [
...
+</ic-tooltip>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcTooltip target="small-chip" label="This is a description">
+  <IcChip id="small-chip" label="Small" size="small">
+    <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
+      <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" />
+    </svg>
+</IcTooltip>`,
```
Comment (jd239) [inline src/content/structured/components/tooltips/code.mdx:166]: There is no closing `</IcChip>`

**Review by jd239 — CHANGES_REQUESTED** [review]
Just a single comment.

File: src/content/structured/components/tooltips/code.mdx [hunk 2]
```diff
@@ -19,20 +19,28 @@ tabs:
...
 
 ## Component demo
 
 export const snippets = [
   {
     language: "Web component",
-    snippet: `<ic-tooltip target="text-button" label="This is a description of the button" placement="top"></ic-tooltip>
-<ic-tooltip target="text-button" label="This is a description of the button" placement="bottom"></ic-tooltip>`,
+    snippet: `<ic-tooltip target="text-button" label="This is a description of the button" placement="top">
+  <ic-button id="text-button"> Top </ic-button>
```
Comment (jd239) [inline src/content/structured/components/tooltips/code.mdx:30]: Can we give the buttons unique IDs please. Perhaps add an index number at the end `text-button-0` and remove the white-spaces around the button labels. I don't think they are required.

**Review by jd239 — CHANGES_REQUESTED** [review]

File: src/content/structured/components/tooltips/code.mdx [hunk 3]
```diff
@@ -19,20 +19,28 @@ tabs:
...
+</ic-tooltip>`,
   },
   {
     language: "React",
-    snippet: `<IcTooltip target="text-button" label="This is a description of the button" placement="top" />
-<IcTooltip target="text-button" label="This is a description of the button" placement="bottom" />`,
+    snippet: `<IcTooltip target="text-button" label="This is a description of the button" placement="top">
+  <IcButton id="text-button"> Top </IcButton>
+</IcTooltip>
+<IcTooltip target="text-button" label="This is a description of the button"  placement="bottom">
```
Comment (jd239) [inline src/content/structured/components/tooltips/code.mdx:41]:
```suggestion
<IcTooltip target="text-button" label="This is a description of the button" placement="bottom">
```

File: src/content/structured/components/tooltips/code.mdx [hunk 4]
```diff
@@ -126,3 +142,52 @@ export const snippetsRight = [
...
+      <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" />
+    </svg>
+  </ic-chip>
+</ic-tooltip>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcTooltip target="small-chip" label="This is a description">
+  <IcChip id="small-chip" label="Small" size="small">
+    <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
```
Comment (jd239) [inline src/content/structured/components/tooltips/code.mdx:163]: Can this be updated to use `<SlottedSVG slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg">`.
The reason for this is if a developer copies this snippet into a typescript project, the `slot` attribute on the SVG tag will display an error. Thanks

**Review by MI6-451 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by jd239 — APPROVED** [review]

**Pull request merged**